engine/src/main/java/org/terasology/particles/functions/ParticleSystemFunction.java

```diff
@@ -22,30 +22,9 @@
  */
 public abstract class ParticleSystemFunction<T> {
     private final int rawDataMask;
-    private final Class<T> component;
 
-    public ParticleSystemFunction(Class<T> component, ParticleDataMask dataMask, ParticleDataMask... dataMasks) {
+    public ParticleSystemFunction(ParticleDataMask dataMask, ParticleDataMask... dataMasks) {
         this.rawDataMask = ParticleDataMask.toInt(dataMask, dataMasks);
-        this.component = component;
-    }
-
```

Comment:
I would double-check if it is save to remove hashCode and Equals completely or if we can stick to something like the rawDataMask field. This may affect how these functions are stored in Lists and Maps.

Reply:
I added those implementations myself and they are no longer needed with the new system, and not possible because I removed the component field. I think we want default behavior for them for now.